components/collapsible-panel/collapsible-panel.js

```diff
@@ -0,0 +1,408 @@
+				}
+				.d2l-collapsible-panel-opener > d2l-icon-custom svg {
+					animation: d2l-collapsible-panel-opener-close var(--d2l-collapsible-panel-arrow-time) ease-in-out;
+				}
+				:host([expanded]) .d2l-collapsible-panel-opener > d2l-icon-custom svg {
+					animation: d2l-collapsible-panel-opener-open var(--d2l-collapsible-panel-arrow-time) ease-in-out;
+				}
+				/* stylelint-disable order/properties-alphabetical-order */
+				@keyframes d2l-collapsible-panel-opener-open {
+					0% { transform: rotate(0deg); }
+					25% { transform: rotate(105deg); animation-timing-function: ease-in-out; }
+					50% { transform: rotate(82deg); animation-timing-function: ease-in-out; }
+					75% { transform: rotate(93deg); animation-timing-function: ease-in-out; }
+					100% { transform: rotate(90deg); animation-timing-function: ease-in-out; }
+				}
+				@keyframes d2l-collapsible-panel-opener-close {
+					0% { transform: rotate(90deg); }
+					25% { transform: rotate(-15deg); animation-timing-function: ease-in-out; }
+					50% { transform: rotate(8deg); animation-timing-function: ease-in-out; }
+					75% { transform: rotate(-3deg); animation-timing-function: ease-in-out; }
+					100% { transform: rotate(0deg); animation-timing-function: ease-in-out; }
+				}
+				/* stylelint-enable */
+			}
+			.d2l-collapsible-panel-divider {
+				border-bottom: 1px solid var(--d2l-color-mica);
+				margin-inline: var(--d2l-collapsible-panel-spacing-inline);
+				opacity: 1;
+			}
+			:host([type=inline]) .d2l-collapsible-panel-divider {
+				opacity: 0;
+			}
+			:host(:not([expanded])) .d2l-collapsible-panel:not(.has-summary) .d2l-collapsible-panel-divider {
+				opacity: 0;
+			}
+			.d2l-collapsible-panel-summary,
+			.d2l-collapsible-panel-content {
+				padding: 0.9rem var(--d2l-collapsible-panel-spacing-inline);
+			}
+			:host([type=inline]) .d2l-collapsible-panel-summary,
+			:host([type=inline]) .d2l-collapsible-panel-content {
+				padding-top: 0;
+			}
+			:host([type="inline"]) .d2l-collapsible-panel-summary {
+				margin-top: -0.3rem;
+			}
+			.d2l-collapsible-panel:not(.has-summary) .d2l-collapsible-panel-summary {
+				display: none;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.expanded = false;
+		this.headingLevel = defaultHeading;
+		this.headingStyle = defaultHeading;
+		this.padding = 'default';
+		this.type = 'default';
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		if (this._intersectionObserver) this._intersectionObserver.disconnect();
+	}
+
+	firstUpdated() {
+		if (typeof(IntersectionObserver) === 'function') {
+			this._intersectionObserver = new IntersectionObserver(entries => {
+				if (!this.expanded) return;
+
+				const entry = entries[0];
+				const element = this.shadowRoot.querySelector('.d2l-collapsible-panel');
+				if (entry.isIntersecting) {
+					element.classList.remove('scrolled');
+				} else {
+					element.classList.add('scrolled');
+				}
```

Comment:
Once there's an `isScrolled` state property, this can obviously be simplified:

```suggestion
				this.isScrolled = !entry.isIntersecting;
```